users/views.py

```diff
@@ -1,3 +1,60 @@
-from django.shortcuts import render
+import requests
+import jwt
 
-# Create your views here.
+from django.http  import JsonResponse
+from django.views import View
+from users.models import User
+
+from freepass import settings
+
+class KakaoAPI:
+    def __init__(self):
+        self.kakao_token_url = "https://kauth.kakao.com/oauth/token"
+        self.kakao_user_url  = "https://kapi.kakao.com/v2/user/me"
+
+    def get_kakao_token(self, auth_code):
+        data = {
+            "grant_type"  : "authorization_code",
+            "client_id"   : settings.KAKAO_APP_KEY,
+            "redirect_uri": settings.REDIRECT_URI,
+            "code"        : auth_code
+        }
+        access_token = requests.post(self.kakao_token_url, data=data).json()['access_token']
+        return access_token
+
+    def get_kakao_user(self, kakao_token):
+        kakao_headers = {"Authorization" : f"Bearer ${kakao_token}"}
+        user_info     = requests.get(self.kakao_user_url, headers=kakao_headers).json()
+        response      = {
+            "kakao_id": user_info['id'],
+            "email"   : user_info["kakao_account"]['email'],
+            "gender"  : user_info["kakao_account"]['gender']
+        }
+        return response
+
+
+class KakaoCallbackView(View):
+    def get(self, request):
+        try:
+            auth_code   = request.GET.get('code')
+            kakao_api   = KakaoAPI()
+            kakao_token = kakao_api.get_kakao_token(auth_code)
+            kakao_user  = kakao_api.get_kakao_user(kakao_token)
+
+            print(type(kakao_token))
+            print(kakao_user)
+
+            user, is_created = User.objects.get_or_create(
+                email    = kakao_user['email'],
+                gender   = kakao_user['gender'],
+                kakao_id = kakao_user['kakao_id']
+            )
```

Comment:
kakao_id를 가지고만 user를 판단하고 그 이후 필요한 데이터를 추가해 주세요.
데이터 추가는 1. get_or_create의 defaults 속성을 이용하거나, 2. is_created를 활용할 수 있습니다.
```suggestion
            user, is_created = User.objects.get_or_create(
                kakao_id = kakao_user['kakao_id']
            )
```